src/Symfony/Component/ClassLoader/ClassCollectionLoader.php

```diff
@@ -116,8 +116,8 @@ public static function load($classes, $cacheDir, $name, $autoReload, $adaptive =
         }
 
         // cache the core classes
-        if (!is_dir(dirname($cache))) {
-            mkdir(dirname($cache), 0777, true);
+        if (!is_dir($cacheDir) && !@mkdir($cacheDir, 0777, true) && !is_dir($cacheDir)) {
+            throw new \RuntimeException(sprintf('Class Collection Loader was not able to create a directory: %s', $cacheDir));
```

Comment:
I'd rather expect  the component to separate exceptions: because case "directory can not be created" and "file not dumped/file not renamed" needs different investigations.